File: data-structures/linear/node/Node.cpp
class Node {
public:
  int val;
  Node *next;

  Node(int val, Node *next = nullptr) {
    this->val = val;
    this->next = next;
  }

  int getValue() { return val; }

  Node *getNext() { return next; }

  void setNext(Node *next) { this->next = next; }
};

File: data-structures/linear/node/Node.ts
class Node {
  private value: number;
  private nextNode: Node | null;

  constructor(value: number, nextNode: Node | null = null) {
    this.value = value;
    this.nextNode = nextNode;
  }

  public getValue(): number {
    return this.value;
  }

  public getNextNode(): Node | null {
    return this.nextNode;
  }

  public setNextNode(nextNode: Node | null): void {
    this.nextNode = nextNode;
  }
}

export { Node };
